source/NVDAObjects/IAccessible/__init__.py

```diff
@@ -1107,12 +1107,12 @@ def _get_presentationalRowNumber(self):
 		index=self.IA2Attributes.get('rowindex')
 		if index is None and isinstance(self.parent,IAccessible):
 			index=self.parent.IA2Attributes.get('rowindex')
+		if index is None:

```

Comment:
I'm ok with this code. Just want to point out that you could avoid adding this block by changing this

```suggestion
		if index is not None:
			Try block
```

Then, the if index is None block below this one also covers this case where the attribute is not defined in IA2Attribs.
However, raising early as you do now, might still be the preferred approach.